pkg/executor/test/ddl/ddl_test.go

```diff
@@ -1026,3 +1026,41 @@ func TestRenameMultiTables(t *testing.T) {
 	tk.MustExec("drop database rename2")
 	tk.MustExec("drop database rename3")
 }
+
+func TestDefShardTables(t *testing.T) {
+	store := testkit.CreateMockStore(t, mockstore.WithDDLChecker())
+
+	tk := testkit.NewTestKit(t, store)
+
+	tk.MustExec("set @@session.tidb_enable_clustered_index = off")
+	tk.MustExec("set @@session.tidb_shard_row_id_bits = 4")
+	tk.MustExec("set @@session.tidb_pre_split_regions = 4")
+	tk.MustExec("use test")
+	tk.MustExec("create table t (i int primary key)")
+	result := tk.MustQuery("show create table t")
+	createSQL := result.Rows()[0][1]
+	expected := "CREATE TABLE `t` (\n  `i` int(11) NOT NULL,\n  PRIMARY KEY (`i`) /*T![clustered_index] NONCLUSTERED */\n) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin /*T! SHARD_ROW_ID_BITS=4 PRE_SPLIT_REGIONS=4 */"
+	require.Equal(t, expected, createSQL)
+
+	// test for manual setup shard_row_id_bits and pre_split_regions
```

Comment:
I added the test for your comment https://github.com/pingcap/tidb/pull/55717/files#r1743352402 , @Benjamin2037